### PR TITLE
Support default clr-namespace when loading XAML from Uri or previewer

### LINKED ISSUE #1375: Remove need for assembly in clr-namespace.
We should be able to specify `clr-namespace` XAML namespaces without the `;assembly=` part now.

In addition, if the assembly part is not supplied, we just throw a `"System.ArgumentOutOfRangeException: Length cannot be less than zero"` exception so it's not obvious what's going wrong.

### PR DESCRIPTION
**Pull request by kekekeks**
Fixes #1375

### CODE REVIEW

File: src/Markup/Avalonia.Markup.Xaml/AvaloniaXamlLoaderPortableXaml.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace Avalonia.Markup.Xaml
     /// <summary>
     /// Loads XAML for a avalonia application.
     /// </summary>
-    public class AvaloniaXamlLoaderPortableXaml
+    public class AvaloniaXamlLoader
```
Comment (grokys) [inline src/Markup/Avalonia.Markup.Xaml/AvaloniaXamlLoaderPortableXaml.cs]: Should also rename the file if we're renaming the class.

File: src/Shared/PlatformSupport/AssetLoader.cs [hunk 2]
```diff
@@ -67,7 +67,20 @@ public bool Exists(Uri uri, Uri baseUri = null)
...
+        /// </summary>
+        /// <param name="uri">The URI.</param>
+        /// <param name="baseUri">
+        /// A base URI to use if <paramref name="uri"/> is relative.
+        /// </param>
+        /// <returns>An assembly (optional) and a stream containing the resource contents.</returns>
+        /// <exception cref="FileNotFoundException">
+        /// The resource was not found.
+        /// </exception>
+        public Tuple<Assembly, Stream> OpenWithAssembly(Uri uri, Uri baseUri = null)
```
Comment (grokys) [inline src/Shared/PlatformSupport/AssetLoader.cs]: Hmm this feels a bit hacky, but I'm not sure what a better way to do it might be. It feels like there should be a parsing step before calling `Open`, but this would be another PR...

Maybe it's just the name? `OpenWithAssembly` sounds like you're _passing_ an assembly to use. Maybe `OpenAndGetAssembly`? Not sure.

Comment (grokys) [inline src/Shared/PlatformSupport/AssetLoader.cs]: Also should we be using `Tuple<>`? `()` value tuples are more modern and you can name the members. I think the problem that we were having with them has gone away.

**Review by grokys — COMMENTED** [review]
Looks good in general. Just not sure about the naming of the new method.

**Comment by grokys** [comment]
Also, failing CI!

File: src/Shared/PlatformSupport/AssetLoader.cs (hunk 2 above)
Comment (kekekeks) [reply src/Shared/PlatformSupport/AssetLoader.cs]: ValueTuple issue with .NET 4.7.1 is still there and there is no workaround.

Comment (amis92) [reply src/Shared/PlatformSupport/AssetLoader.cs]: Isn't it fixed? https://github.com/dotnet/standard/issues/567#issuecomment-356500745

**Review by grokys — APPROVED** [review]
I finished this off, hope I did it all right...
